### PR TITLE
Removes obsolete installer sets on upgrade

### PR DESCRIPTION
**Pull request by sm43**
Signed-off-by: Shivam Mukhade <smukhade@redhat.com>

# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Run `make test lint` before submitting a PR
- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/operator/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

### CODE REVIEW

File: pkg/reconciler/kubernetes/tektoninstallerset/client/obsolete_sets.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+	"k8s.io/apimachinery/pkg/selection"
+)
+
+func (i *InstallerSetClient) RemoveObsoleteSets(ctx context.Context) error {
+	var sets []string
+	if i.resourceKind == v1alpha1.KindTektonPipeline {
+		sets = []string{"pipeline", "PrePipeline", "PostPipeline"}
+	} else if i.resourceKind == v1alpha1.KindTektonTrigger {
+		sets = []string{"trigger"}
+	}
```
Comment (savitaashture) [inline pkg/reconciler/kubernetes/tektoninstallerset/client/obsolete_sets.go:34]: Should we do it for all InstallerSets :thinking: i.e Dashboard, Rbac etc...

Comment (sm43) [reply pkg/reconciler/kubernetes/tektoninstallerset/client/obsolete_sets.go:34]: those reconciler are not yet refactored.. once they are done we can do it for them

**Review by savitaashture — APPROVED** [review]

**Review by vdemeester — APPROVED** [review]

**Comment by vdemeester** [comment]
@sm43 can you squash your commit ? (aka no merge commit 😝)

**Pull request merged**